### PR TITLE
[BugFix][Quantization] Recover static W8A8 weight scales for MLAPO

### PR DESCRIPTION
**Pull request by Yuli-yx**
### What this PR does / why we need it?

Static ModelSlim W8A8 checkpoints may store `deq_scale` and `input_scale` without `weight_scale`. The ordinary quantized matmul uses the loaded `deq_scale`, but MLAPO reads the uninitialized `weight_scale` parameter. This produces garbled output with Kimi K2.6 W4A8 in two-node PD serving. This checkpoint uses W4A8_DYNAMIC for MoE experts and static W8A8 for attention projections, which are supported by MLAPO.

Recover FP32 per-channel weight scales from `deq_scale / input_scale` after loading, decoding the low FP32 bits for FP16 checkpoints. Keep the existing compressed-tensors path, whose checkpoint supplies weight scales.

### Does this PR introduce _any_ user-facing change?

MLAPO returns coherent output for static ModelSlim checkpoints that omit independent weight scales. No configuration or API changes.

### How was this patch tested?

- Reproduced garbled output without DFlash using the source from [nightly run 34766527833](https://github.com/vllm-project/vllm-ascend/actions/runs/34766527833), vLLM v0.28.0 and vLLM-Ascend `660c4582` on both A3 nodes. Producer TP8/DP2, consumer TP4/DP4, expert parallelism, fused MC2, consumer FULL_DECODE_ONLY, max model length 16384.
- Direct producer generation answered `17 + 25 = 42`; PD serving produced repeated numbers and mixed characters. Disabling only consumer MLAPO restored coherent PD output. All 244 static attention projections in the checkpoint lack `weight_scale` and supply `deq_scale`.
- Added regression coverage for missing scale tensors with BF16 and encoded FP16 dequantization scales, checking recovered values and preservation of `deq_scale`. NPU int64-to-int32 FP32 bitcast verified separately.
- CPU/mock preflight uses the checked-in `vllm-ascend/.venv`; the isolated source worktree is selected through PYTHONPATH:

```bash
cd /Users/zhuyixiang/Projects/vllm-workspace/vllm-ascend
PYTHONPATH=/Users/zhuyixiang/Projects/vllm-workspace/.vaws-local/kimi-k26-accuracy-20260916/source uv run pytest -q /Users/zhuyixiang/Projects/vllm-workspace/.vaws-local/kimi-k26-accuracy-20260916/source/tests/ut/quantization/methods/w8a8/test_w8a8_static.py /Users/zhuyixiang/Projects/vllm-workspace/.vaws-local/kimi-k26-accuracy-20260916/source/tests/ut/quantization/test_method_adapters.py /Users/zhuyixiang/Projects/vllm-workspace/.vaws-local/kimi-k26-accuracy-20260916/source/tests/ut/quantization/test_weight_switch.py
```

Result: **32 passed, 2 dtype subtests passed**. All applicable pre-commit hooks passed.

- With the patch applied on both nodes and MLAPO enabled, the original failed PD request returns `17 plus 25 equals 42`, identical to the MLAPO-disabled control.
- **Full GSM8K test split: 1278/1319 correct, 96.8916% accuracy** with AISBench zero-shot CoT, temperature 0, top_p 1, top_k -1, seed 1024, batch size 32, max output 8192, stop string `Question`. Dataset question/answer records and order verified against the canonical OpenAI 1319-item test split. Reasoning is removed before GSM8K numeric answer postprocessing.
- All **1319 requests succeeded (HTTP 200)**; raw response IDs were matched back to all 1319 dataset items. **1304 normal stops and 15 length stops**; all capped samples are included in the full accuracy result (14 incorrect, 1 correct). Raw response checks found no empty outputs, replacement characters, excessive digit/mixed-character output or repeated-line anomalies. Reviewed incorrect-answer tails remain coherent mathematical reasoning; this test does not establish token-for-token equivalence with ordinary MLA.
- DFlash was disabled throughout this reproduction and fixed accuracy run; its separate failure is outside this patch. All inference requests originated on the servers.

- vLLM main: https://github.com/vllm-project/vllm/commit/84030bbe3d74d99bad477a3d2e37a973ccd8865c

### CODE REVIEW

File: vllm_ascend/quantization/methods/w8a8/w8a8_static.py [hunk 1]
```diff
@@ -135,3 +135,13 @@ def process_weights_after_loading(self, layer):
         if self.quant_method == COMPRESSED_TENSORS_METHOD:
             deq_scale = layer.input_scale.data * layer.weight_scale.data
             layer.deq_scale = torch.nn.Parameter(deq_scale, requires_grad=False)
+        else:
+            # Static ModelSlim checkpoints need not store weight_scale. The
+            # regular quantized matmul uses deq_scale, but MLAPO consumes the
+            # per-channel weight scale directly. Recover it from the loaded
+            # activation/weight product instead of reading an empty parameter.
+            deq_scale = layer.deq_scale.data
```
Comment (SparrowMu) [inline vllm_ascend/quantization/methods/w8a8/w8a8_static.py]: need to check if weight_scale is not empty and if deq_scale exists and is not empty

**Comment by czydyy** [comment]
/nightly Kimi-K2.6-W4A8-64k-1k-TPOT50-PD
nightly command triggered.
- Nightly-A3: [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/35220025438)

**Review by linfeng-yuan — APPROVED** [review]
